### PR TITLE
docs(material/theming): update guide for density

### PR DESCRIPTION
**Pull request by jelbourn**

### CODE REVIEW

File: guides/theming.md [hunk 1]
```diff
@@ -375,6 +382,38 @@ $my-palette: mat.define-palette(mat.$indigo-palette);
...
+density system. The [Material Design density guidance](https://m2.material.io/design/layout/applying-density.html)
+explicitly discourages increasing density for such interactions because they don't compete for space in the
+application's layout.
+
+You can apply custom density setting to the entire library or to individual components using their density Sass mixins.
+
+```scss
+// You can set a density setting in your theme to apply to all components.
+$dark-theme: mat.define-dark-theme((
+ color: ...,
```
Comment (mmalerba) [inline guides/theming.md:406]: nit: indent one more space

**Review by mmalerba — APPROVED** [review]
LGTM

**Review by crisbeto — APPROVED** [review]
LGTM

File: guides/theming.md [hunk 2]
```diff
@@ -375,6 +382,38 @@ $my-palette: mat.define-palette(mat.$indigo-palette);
...
+## Customizing density
+
+Angular Material's density customization is based on the
+[Material Design density guidelines](https://m2.material.io/design/layout/applying-density.html). This system
+defines a scale where zero represents the default density. You can decrement the number for _more density_ and increment
+the number for _less density_.
+
+The density system is based on a *density scale*. The scale starts with the
+default density of `0`. Each whole number step down (`-1`, `-2`, etc.) reduces
+the affected sizes by `4px`, down to the minimum size necessary for a component to render coherently.
```
Comment (twerske) [inline guides/theming.md:394]: Whole step down, what about whole step up? Or should this be negative values only (<0)?

**Review by twerske — APPROVED** [review]
lgtm

File: guides/theming.md (hunk 2 above)
Comment (jelbourn) [reply guides/theming.md:394]: This is only negative values for now.

**Pull request merged**